sycl/source/detail/graph_impl.cpp

```diff
@@ -296,6 +299,183 @@ void exec_graph_impl::makePartitions() {
   }
 }
 
+#if SYCL_EXT_CODEPLAY_KERNEL_FUSION
+void exec_graph_impl::fuseGraph() {
+  if (!MEnableFusion && !MRequireFusion) {
+    return;
+  }
+
+  bool SupportGraphFusion =
+      MGraphImpl->getDevice().has(aspect::ext_oneapi_graph_fusion);
+
+  if (!SupportGraphFusion || MIsUpdatable || !singleDevicePartitions()) {
+    if (MRequireFusion) {
+      std::string Cause;
+      if (!SupportGraphFusion) {
+        Cause = "Graph Fusion not supported.";
+      } else if (MIsUpdatable) {
+        Cause = "Updatable Graph cannot be fused.";
```

Comment:
Is this behaviour documented in the sycl_ext_oneapi_graph_fusion extension spec? Should it be given that the spec is experimental we can always update it once supported. Same comment for host-task.

I'd say this is more relevant than the current "Multi-device graph" limitation, given that multi-device graphs aren't currently possible.